ext/opentelemetry-ext-flask/src/opentelemetry/ext/flask/__init__.py

```diff
@@ -158,11 +159,20 @@ class FlaskInstrumentor(BaseInstrumentor):
 
     def __init__(self):
         super().__init__()
-        self._original_flask = None
+        self._original_flask_class = None
 
-    def _instrument(self):
-        self._original_flask = flask.Flask
+    def _instrument(
+        self, flask_class=None
+    ):  # pylint: disable=arguments-differ
+        if flask_class is not None:
```

Comment:
This `if` makes this function to behave very different. Related to my long comment above, the case where you pass `flask_class` is not doing any instrumentation but just returning `_InstrumentedFlask`.

Reply:
I think we can find another way of using `_instrument` here. Maybe pass the Flask app as was done before? That would be fine. I still would like to have instrumentation depend on it being done before something is imported only as a very last option because of how fragile and hard to debug this approach is (it also breaks PEP8).

Reply:
What is the reason to implement extra functionalities in `_instrument()`?,  I think the scope of this method is to be called by the autoinstrument command and  by users that want to enable instrumentation in all that framework. If we start adding extra functionalities and special cases we'll end up with a lot of different `_instrument()` what must be used in a specify way to work.

What do you think about exposing InstrumentedFlask to the user for the "programmatic instrumentation" case as I exposed before?